### DISCUSSION TITLE
How to use this with a simple project

### DISCUSSION THREAD

**Discussion by Nneji123** [fief-dev/fief#317] [Q&A] [answered]
Supposing I'd want to build a simple fastapi api(e.g a post and comment app) and I need to relate the user that's authenticated by fief.dev to a post and comment how do I go about that?

**Comment by fief-bailiff** [comment]
Hail, @Nneji123 👋 Welcome to Fief's kingdom!

Our team will get back to you very soon to help.

In the meantime, take a minute to star our repository ⭐️

![star-fief](https://user-images.githubusercontent.com/1144727/192523552-dc2c31fa-ae94-42fa-9ab4-7d0ba51bb7b0.png)

Farewell!

**Comment by frankie567** [comment, marked as answer]
You can refer to the following example: https://docs.fief.dev/integrate/python/fastapi/#api-example

```py
@app.get("/user")
async def get_user(
    access_token_info: FiefAccessTokenInfo = Depends(auth.authenticated()),  
):
    return access_token_info
```

[`FiefAccessTokenInfo`](https://fief-dev.github.io/fief-python/fief_client.html#FiefAccessTokenInfo) is a dictionary containing the `id` key; which is the ID of the user in Fief. You can use this ID to reference the user in your app and data.

**Reply by Nneji123** [reply]
Would that mean in the code below I'd have to also create a user table to form the relationship? Or is there another way to go about it?:

```python
# Todos table
class Todo(Base):
    __tablename__ = 'todos'

    todo_id = Column(Integer, primary_key=True)
    user_id = Column(Integer, ForeignKey('users.user_id'))  # Reference to users table
    task = Column(String(100), nullable=False)
    description = Column(Text)
    completed = Column(Boolean, default=False)

    # Relationship to the Users table
    owner = relationship("User", back_populates="todos")


# Define the User table as it exists in your extern
class User(Base):
    __tablename__ = 'users'

    user_id = Column(Integer, primary_key=True)
    email = Column(String(100), unique=True, nullable=False)
    # Other user-related fields...

    # Relationship to the Todos table
    todos = relationship("Todo", back_populates="owner")
```

**Reply by frankie567** [reply]
Well, you can do that, but since Fief stores the user data itself, you don't have to have a dedicated user table. You can juste store `user_id` as a plain UUID in your `Todo` model (so, without any foreign key), knowing that it exists in Fief.
